### PR TITLE
Add StartRideMusic field to RTD

### PR DESCRIPTION
**Pull request by frutiemax**
Circus has a special behavior: every ride but the circus uses DefaultStartRideMusicChannel.

### CODE REVIEW

**Comment by IntelOrca** [comment]
If it is only circus that has different behaviour, I would not add a set field to every other ride type. You can set defaults in the `RideTypeDescriptor` struct.

**Comment by frutiemax** [comment]
> If it is only circus that has different behaviour, I would not add a set field to every other ride type. You can set defaults in the `RideTypeDescriptor` struct.

How do you set defaults?

**Comment by IntelOrca** [comment]
```
struct X
{
    int num = 4;
};
```

**Comment by Gymnasiast** [comment]
@frutiemax Can you rebase/remake this? There have been some changes in our codebase in the meantime.

File: src/openrct2/ride/RideAudio.cpp [hunk 1]
```diff
@@ -173,40 +159,44 @@ namespace OpenRCT2::RideAudio
...
+
+            _musicChannels.emplace_back(instance, channel);
+        }
+    }
+
+    static void StartRideMusicChannel(const ViewportRideMusicInstance& instance)
+    {
+        // Create new music channel
+        auto ride = get_ride(instance.RideId);
+        const auto& rtd = GetRideTypeDescriptor(ride->type);
```
Comment (Gymnasiast) [inline src/openrct2/ride/RideAudio.cpp:197]: ride->GetRideTypeDescriptor()

**Review by Gymnasiast — APPROVED** [review]

**Comment by frutiemax** [comment]
A bit of a pain to merge but it's tested and working...

**Pull request merged**